### PR TITLE
LogServices

### PR DESCRIPTION
**Pull request by ConfusedSatlan**

### CODE REVIEW

File: src/main/java/mate/academy/Main.java [hunk 1]
```diff
@@ -6,15 +6,19 @@
...
+    private static final Logger logger = LogManager.getLogger(Main.class);
+
     public static void main(String[] args) {
         AuthenticationService authenticationService = new AuthenticationServiceImpl();
         User user;
         try {
             user = authenticationService.login("bob", "1234");
         } catch (AuthenticationException e) {
-            e.printStackTrace();
+            logger.error("Authentication failed: " + e.getMessage());
```
Comment (vovasalyha) [inline src/main/java/mate/academy/Main.java:21]:
```suggestion
            logger.error("Authentication failed", e);
```

File: src/main/java/mate/academy/service/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -2,11 +2,15 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 public class AuthenticationServiceImpl implements AuthenticationService {
+    private static final Logger logger = LogManager.getLogger(AuthenticationServiceImpl.class);
+
     @Override
     public User login(String login, String password) throws AuthenticationException {
-        //TODO: add corresponding log message about method login was called
+        logger.info("AuthenticationServiceImpl login method is called. Login: {}", login);
```
Comment (vovasalyha) [inline src/main/java/mate/academy/service/AuthenticationServiceImpl.java:13]: Logging that some method was called is more suitable for debug level, as usually you don't want to overwhelm your logs with unnecessary information.

File: src/main/java/mate/academy/service/OrderServiceImpl.java [hunk 3]
```diff
@@ -2,13 +2,18 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 public class OrderServiceImpl implements OrderService {
+    private static final Logger logger = LogManager.getLogger(OrderServiceImpl.class);
+
     @Override
     public Order completeOrder(Long userId) {
-        // TODO: add log message about method completeOrder was called
+        logger.info("OrderServiceImpl completeOrder method is called. UserId: {}", userId);
```
Comment (vovasalyha) [inline src/main/java/mate/academy/service/OrderServiceImpl.java:16]: Same here.

File: src/main/java/mate/academy/service/OrderServiceImpl.java [hunk 4]
```diff
@@ -24,7 +29,12 @@ private List<Product> getAllProductsFromShoppingCart(Long userId) {
         Product macBook = new Product("MacBook Air 2020", BigDecimal.valueOf(1399));
         Product xiaomi = new Product("Xiaomi 12", BigDecimal.valueOf(499));
         List<Product> products = List.of(iphone, macBook, xiaomi);
-        // TODO: add log message about successful fetched data from DB
+        logger.info("OrderServiceImpl getAllProductsFromShoppingCart method is done "
+                + "for userId: {}, productList is: {}",
+                userId, products.stream()
```
Comment (vovasalyha) [inline src/main/java/mate/academy/service/OrderServiceImpl.java:34]: Why do you need so complex computation just to write message to log? A lot of such logs can decrease your performance badly.

File: src/main/resources/log4j2.xml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        <Logger name="mate.academy" level="info" additivity="false">
+            <AppenderRef ref="LogToFile"/>
+            <AppenderRef ref="LogToConsole"/>
+        </Logger>
+        <Root level="error">
+            <AppenderRef ref="LogToFile"/>
+            <AppenderRef ref="LogToConsole"/>
+        </Root>
+    </Loggers>
+</Configuration>
```
Comment (vovasalyha) [inline src/main/resources/log4j2.xml:23]: image

**Review by vovasalyha — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/Main.java (hunk 1 above)
Comment (nacenik) [reply src/main/java/mate/academy/Main.java:21]: not fixed

File: .gitignore [hunk 6]
```diff
@@ -1,3 +1,4 @@
 .idea/*
 *.iml
 target/*
+logs/
```
Comment (nacenik) [inline .gitignore:4]:
```suggestion
logs/*
```

**Review by nacenik — APPROVED** [review]
Good solution, you can improve it if you want :)

**Pull request closed without merging**